### PR TITLE
[TEST] Activate automatic gcov report on SonarCloud

### PR DESCRIPTION
**Pull request by kathvargasr**
Sensor Generic Coverage Report on SonarCloud does not recognize the paths generated by custom target code-coverage.

More precisely, when running gcovr, paths are smoothen to its source directory. 

So by running gcovr through a custom target, we'll always have relative paths, and Sonar's coverage sensor only recognizes absolute paths

To bypass this issue, we're now running gcovr directly on workflow.

### CODE REVIEW

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -128,9 +129,9 @@ jobs:
           cd $GITHUB_WORKSPACE/_build
           ctest -C Release --output-on-failure -L "unit"
 
-      - name: Compile coverage reports
+      - name: Collect coverage into one XML report
         run: |
-          cmake --build $GITHUB_WORKSPACE/_build --target code-coverage
+          gcovr --sonarqube > coverage.xml
```
Comment (flomnes) [inline .github/workflows/sonarcloud.yml]: From the doc
```suggestion
          gcovr --sonarqube --output coverage.xml
```

File: sonar-project.properties [hunk 2]
```diff
@@ -12,7 +12,8 @@ sonar.sources=src
 sonar.sourceEncoding=UTF-8
 
 sonar.exclusions=src/ext/**
+sonar.coverage.exclusions=src/ext/**,src/tests/**,src/analyzer/**,src/distrib/**,src/solver/**,src/tools/**,src/ui/**
```
Comment (flomnes) [inline sonar-project.properties]: Don't exclude **src/solver**. The rest is OK.

File: .github/workflows/sonarcloud.yml [hunk 3]
```diff
@@ -85,6 +85,7 @@ jobs:
         with:
           antares-deps-version: ${{steps.antares-deps-version.outputs.prop}}
           os: ${{matrix.os}}
+          buildtype: Debug
```
Comment (flomnes) [inline .github/workflows/sonarcloud.yml:88]: unused, consider removing

**Review by flomnes — CHANGES_REQUESTED** [review]

File: .github/workflows/sonarcloud.yml (hunk 1 above)
Comment (kathvargasr) [reply .github/workflows/sonarcloud.yml]: The example for cfamily from the docs suggested the > but we can try the option and see if it doesn't smooth the paths wrongly

Comment (flomnes) [reply .github/workflows/sonarcloud.yml]: From `man gcovr`
```
       -o OUTPUT, --output OUTPUT
              Print output to this filename. Defaults to stdout.  Required for
              --html-details.
```
So redirection from stdout to `coverage.xml` and this option should be equivalent.

File: .github/workflows/sonarcloud.yml (hunk 3 above)
Comment (kathvargasr) [reply .github/workflows/sonarcloud.yml:88]: seems that dependencies cannot be built correctly without this. cmake fails with:

```
CMake Error at CMakeLists.txt:249 (message):
  Sirius solver not found.  Sirius solver can be compiled with
  -DBUILD_sirius=ON or you can specify previous dependency install directory
  with -DCMAKE_PREFIX_PATH or -DDEPS_INSTALL_DIR
```

Comment (flomnes) [reply .github/workflows/sonarcloud.yml:88]: you are right, sorry

**Review by flomnes — APPROVED** [review]

**Pull request merged**